src/gui/torrentcontentmodel.cpp

```diff
@@ -88,6 +98,43 @@ namespace
             return QIcon(iconPixmap);
         }
     };
+#elif defined(Q_OS_MAC)
+    // There is a similar bug on macOS, to be reported to Qt
+    // https://github.com/qbittorrent/qBittorrent/pull/6156#issuecomment-316302615
+    class MacFileIconProvider: public UnifiedFileIconProvider
+    {
+    public:
+        using QFileIconProvider::icon;
+        QIcon icon(const QFileInfo &info) const override
+        {
+            QString suffix = info.suffix();
```

Comment:
Well, it is discussible due to the size of the function indeed. But your project — your rules. Done :)